bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/ProfileTypeUID.java

```diff
@@ -47,4 +49,23 @@ public String getId() {
         return getSegment(1);
     }
 
+    @Override
+    public boolean equals(@Nullable Object obj) {
```

Comment:
Why is that? This class inherits the `equals()` implementation from `AbstractUID`, which essentially does the same. Or did I miss anything?